packages/medusa/src/services/product-category.ts

```diff
@@ -65,6 +65,38 @@ class ProductCategoryService extends TransactionBaseService {
 
     return productCategoryTree
   }
+
+  /**
+   * Deletes a product category
+   *
+   * @param productCategoryId is the id of the product category to delete
+   * @return a promise
+   */
+  async delete(productCategoryId: string): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const productCategoryRepository: ProductCategoryRepository =
+        manager.getCustomRepository(this.productCategoryRepo_)
+
+      const productCategory = await this.retrieve(productCategoryId, {
+        relations: ["category_children"],
+      })
```

Comment:
todo: Retrieve will throw on non-existing categories, so to resolve idempotently we should catch the error and return undefined
```suggestion
      const productCategory = await this.retrieve(productCategoryId, {
        relations: ["category_children"],
      }).catch(err => void 0)
```

Reply:
.catch(() => void 0)